pkg/azure/ipam/instances.go

```diff
@@ -87,64 +88,117 @@ func (m *InstancesManager) Resync(ctx context.Context) time.Time {
 func (m *InstancesManager) resyncInstance(ctx context.Context, instanceID string) time.Time {
 	resyncStart := time.Now()
 
-	vnets, subnets, err := m.api.GetVpcsAndSubnets(ctx)
-	if err != nil {
-		m.logger.Warn("Unable to synchronize Azure virtualnetworks list", logfields.Error, err)
-		return time.Time{}
-	}
-
-	instance, err := m.api.GetInstance(ctx, subnets, instanceID)
+	// First get the instance with empty subnet map to extract subnet IDs
+	instance, err := m.api.GetInstance(ctx, ipamTypes.SubnetMap{}, instanceID)
 	if err != nil {
 		m.logger.Warn("Unable to synchronize Azure instance interface list",
 			logfields.Error, err,
 			logfields.InstanceID, instanceID,
 		)
 		return time.Time{}
 	}
+	
+	// Extract subnet IDs from this instance
+	instanceMap := ipamTypes.NewInstanceMap()
+	instanceMap.UpdateInstance(instanceID, instance)
+	nodeSubnetIDs := m.extractSubnetIDs(instanceMap)
+	
+	// Query targeted subnets
+	subnets, err := m.api.GetNodesSubnets(ctx, nodeSubnetIDs)
+	if err != nil {
+		m.logger.Warn("Unable to synchronize Azure subnets list for instance", 
+			logfields.Error, err, logfields.InstanceID, instanceID)
+		// Continue with empty subnets map rather than failing completely
+		subnets = ipamTypes.SubnetMap{}
+	}
+	
+	// Re-query instance with discovered subnets for complete information
+	if len(subnets) > 0 {
+		instance, err = m.api.GetInstance(ctx, subnets, instanceID)
+		if err != nil {
+			m.logger.Warn("Unable to re-synchronize Azure instance with subnet details",
+				logfields.Error, err,
+				logfields.InstanceID, instanceID,
+			)
+			return time.Time{}
+		}
+	}
 
 	m.logger.Info(
-		"Synchronized Azure IPAM information for the corresponding instance",
+		"Synchronized Azure IPAM information for the corresponding instance using targeted subnet discovery",
 		logfields.InstanceID, instanceID,
-		logfields.NumVirtualNetworks, len(vnets),
 		logfields.NumSubnets, len(subnets),
+		"targeted_subnets", len(nodeSubnetIDs),
 	)
 
 	m.mutex.Lock()
 	defer m.mutex.Unlock()
 	m.instances.UpdateInstance(instanceID, instance)
-	m.vnets = vnets
 	m.subnets = subnets
 
 	return resyncStart
 }
 
-// resyncInstances performs a full sync of all instances
+// extractSubnetIDs extracts unique subnet IDs from node network interfaces
+func (m *InstancesManager) extractSubnetIDs(instances *ipamTypes.InstanceMap) []string {
+	subnetIDs := make(map[string]struct{})
+	
+	instances.ForeachAddress("", func(instanceID, interfaceID, ip, poolID string, address ipamTypes.Address) error {
+		if poolID != "" {
```

Comment:
can you share offical azure doc where it says poolID is ARM resource id of subnet?